source/material_model/visco_plastic.cc

```diff
@@ -228,9 +228,6 @@ namespace aspect
                                                                                    phase_function_values,
                                                                                    phase_function.n_phase_transitions_for_each_composition());
```

Comment:
You could also move these statements into their respective `switch` clauses below so that they don't get executed if you don't actually need these factors.